crates/solvers/src/domain/order.rs

```diff
@@ -94,10 +94,15 @@ pub enum Class {
 /// the other hand, liquidity orders are CoW Protocol orders, meaning that they
 /// first provide the tokens being swapped to and only get paid at the end of
 /// the settlement.
+// TODO I'd still prefer stronger types. Something like
+// pub enum Order { Liquidity(Liquidity), Market(Market), Limit(Limit) }
+// pub enum UserOrder { Market(Market), Limit(Limit) }
 #[derive(Debug)]
 pub struct NonLiquidity<'a>(&'a Order);
 
 impl<'a> NonLiquidity<'a> {
+    // TODO Should this be called UserOrder? The term "user order" is used
+    // here
```

Comment:
I think that is a better name.

Reply:
https://github.com/cowprotocol/services/pull/1513/commits/346b2300c078e82d26f819c0182d359bd3c3d3e3